pxr/usdValidation/usdUtilsValidators/testenv/testUsdUtilsValidators.cpp

```diff
@@ -160,12 +162,125 @@ TestFileExtensionValidator()
     TF_AXIOM(errors.empty());
 }
 
+static void
+TestRootPackageValidator()
+{
+    UsdValidationRegistry &registry = UsdValidationRegistry::GetInstance();
+
+    // Verify the validator exists
+    const UsdValidationValidator *validator = registry.GetOrLoadValidatorByName(
+        UsdUtilsValidatorNameTokens->rootPackageValidator);
+
+    TF_AXIOM(validator);
+
+    // Load the pre-created usdz stage with a compressed file in the root layer
+    {
+        const UsdStageRefPtr &stage =
+            UsdStage::Open("packageWithCompressionAndByteAlignmentErrors.usdz");
+        const UsdValidationErrorVector errors = validator->Validate(stage);
+
+        // Verify compression error occurs
+        TF_AXIOM(errors.size() == 3);
+
+        const std::vector<std::string> expectedErrorMessages = {
+            "File 'test.usda' in package "
+            "'packageWithCompressionAndByteAlignmentErrors.usdz' has an "
+            "invalid offset 39.",
+            "File 'texture.jpg' in package "
+            "'packageWithCompressionAndByteAlignmentErrors.usdz' has "
+            "compression. Compression method is '8', actual size is 14. "
+            "Uncompressed size is 1028.",
+            "File 'texture.jpg' in package "
+            "'packageWithCompressionAndByteAlignmentErrors.usdz' has an "
+            "invalid offset 131."};
+        const std::vector<TfToken> expectedErrorTokens = {
+            TfToken("usdUtilsValidators:RootPackageValidator.ByteMisalignment"),
+            TfToken("usdUtilsValidators:RootPackageValidator.CompressionDetected"),
+            TfToken("usdUtilsValidators:RootPackageValidator.ByteMisalignment")};
+
+        for (size_t i = 0; i < errors.size(); ++i)
+        {
+            ValidateError(errors[i], expectedErrorMessages[i],
+                expectedErrorTokens[i]);
+        }
+    }
+
+    // Load the pre-created usdz stage with valid files
+    {
+        const UsdStageRefPtr &stage =
+            UsdStage::Open("pass.usdz");
+        const UsdValidationErrorVector errors = validator->Validate(stage);
+
+        // Verify there are no errors
+        TF_AXIOM(errors.empty());
+    }
+}
+
+static void
+TestUsdzPackageValidator()
+{
+    UsdValidationRegistry &registry = UsdValidationRegistry::GetInstance();
+
+    // Verify the validator exists
+    const UsdValidationValidator *validator = registry.GetOrLoadValidatorByName(
+        UsdUtilsValidatorNameTokens->usdzPackageValidator);
+
+    TF_AXIOM(validator);
+
+    // Load the pre-created usdz stage with a compressed file in the root layer
+    {
+        const UsdStageRefPtr &stage =
+            UsdStage::Open("sceneWithInvalidPackage.usda");
+
+        const UsdValidationErrorVector errors = validator->Validate(stage);
+
+        // Verify internal compression and byte misalignment errors occur
+        TF_AXIOM(errors.size() == 3);
+
+        const std::string fullErrorPath = ArchGetCwd() +
```

Comment:
@tallytalwar Two things about this line. 
1. It seems in the validator itself, if I pass in a root level myFile.usdz, the `ArSplitPackageRelativePathOuter` will get the package path that i want, as in, only `myFile.usdz` however passing in `parentFile.usda` that references `myFile.usdz` the error message has the full real path in the error message
2. If the above is okay as is, I wanted to join these paths here in a platform agnostic way and I thought there was some library function to join these paths so it would work everywhere as this as written seems it's the wrong direction for Windows, and it seemed wrong to write some helper function here to deal with the correct slashes. I did find other tests that used only this slash, but let me know if this should be changed.

Reply:
I am not sure I fully understand, 1.

But for 2, there is: https://github.com/PixarAnimationStudios/OpenUSD/blob/dev/pxr/base/tf/stringUtils.h#L677-L689 which you can use to do a platform agnostic way of concatenating paths, etc.

Reply:
I see, yeah I think I can reason about that (will also confirm internally), but the reference path to internal package gets fully resolved relative to its referenced location, in order to locate the asset, and that causes the paths to be absolute.